File: sol/solution/0001-0100/0007. Reverse Integer/Solution.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut num = x;
        let mut rev: i32 = 0;
        while num != 0{
            let digit = num%10;
            if rev>i32::MAX / 10 || (rev == i32::MAX){
                return 0;
            }
            if rev<i32::MIN / 10 || (rev == i32::MIN){
                return 0;
            }
            rev = (rev*10)+digit;
            num = num/10;
        }
        rev
    }
}

File: sol/solution/0001-0100/0007. Reverse Integer/Solution.scala
object Solution {
    def reverse(x: Int): Int = {
        var n = x
        var rev = 0
        while (n != 0) {
            val pop = n % 10
            n /= 10
            if (rev > Int.MaxValue / 10 || (rev == Int.MaxValue / 10 && pop > 7)) return 0
            if (rev < Int.MinValue / 10 || (rev == Int.MinValue / 10 && pop < -8)) return 0
            rev = rev * 10 + pop
        }
        rev
    }
}